### PR TITLE
[rv_core_ibex] Add rv_core_ibex dedicated peripheral 

### PR DESCRIPTION
**Pull request by tjaychen**
The peripheral will eventually be responsible for the following items

- alert generation and hookup
- configuration for simple address translation scheme
- error bit vectoring

Note, long term we should add functionality to template hosts as well as devices.  
Once that is done, there will no longer be a need for such templated peripherals.

### CODE REVIEW

File: hw/ip/rv_core_ibex_peri/data/rv_core_ibex_peri.hjson
Comment (sriyerg) [inline hw/ip/rv_core_ibex_peri/data/rv_core_ibex_peri.hjson:1]: Is there a downside of maintaining these sources in `hw/ip/rv_core_ibex` itself?

File: hw/Makefile
Comment (sriyerg) [inline hw/Makefile:36]: Given that this is a TL "device" that contains CSRs and is not actually a peripheral such as UART / SPI, how about `rv_core_ibex_csrs`? It also makes it clear that we are maintaining CSRs that would otherwise be a part of `rv_core_ibex` in this separate module.

File: hw/ip/rv_core_ibex_peri/data/rv_core_ibex_peri.hjson
Comment (tjaychen) [reply hw/ip/rv_core_ibex_peri/data/rv_core_ibex_peri.hjson:1]: no, we can move it over there if you prefer.

File: hw/top_earlgrey/data/top_earlgrey.hjson
Comment (sriyerg) [inline hw/top_earlgrey/data/top_earlgrey.hjson:621]: I suppose instantiating this in the top rather than inside `rv_core_ibex` is easier from tooling / automation point of view, at the expense of increased integration DV impact?

Comment (sriyerg) [reply hw/top_earlgrey/data/top_earlgrey.hjson:621]: Ooh, I take that back - the DV impact is the same because `rv_core_ibex` is not verified. Ok with instantiating this at the top if that's easier.

File: hw/Makefile
Comment (tjaychen) [reply hw/Makefile:36]: it also currently contains the alert senders, so slightly more than just csrs

File: hw/top_earlgrey/data/top_earlgrey.hjson
Comment (tjaychen) [reply hw/top_earlgrey/data/top_earlgrey.hjson:621]: yeah mostly because there's a lot of automation stuff missing. We don't currently the hosts, we would have to add that in so that we auto-wire tlul, alerts, interrupts etc. 

So this is just a cop-out solution to fit into our existing infrastructure.

**Comment by sriyerg** [comment]
LGTM, Thanks.

**Comment by tjaychen** [comment]
i'm going to hold off on merging this because michael is looking into
whether it's simple to just template the host.
i kind of punted on that, but doesn't mean it can't be done.

either way though, i will move the code to the rv_core_ibex area as you
suggested instead of having a new rv_core_ibex_peri directory.

**Comment by tjaychen** [comment]
actually on second though, let me merge this in first. Because i want to have a baseline before i start templating, otherwise the PR is going to look too big. 

could someone quickly approve this and i'll rebase and merge?

**Review by msfschaffner — APPROVED** [review]
LGTM, thanks for adding this.

Btwy, @weicaiyang and I could sort out the DV issues on #7129, so once that one has landed, it should be possible to make comportable modules with both `host` and `device` TL-UL plugs.

**Comment by tjaychen** [comment]
sounds good. do you mind if i get this one in first. And then i'll follow yours to template rv_core_ibex?

**Comment by msfschaffner** [comment]
no, not at all. please go ahead :+1: .

**Review by tomroberts-lowrisc — APPROVED** [review]

**Pull request merged**
